sdk/python/kfp/containers/_k8s_job_helper.py

```diff
@@ -48,24 +48,10 @@ def _create_k8s_job(self, yaml_spec):
                                                             annotations=yaml_spec['metadata']['annotations']))
     container = k8s_client.V1Container(name = yaml_spec['spec']['containers'][0]['name'],
                                        image = yaml_spec['spec']['containers'][0]['image'],
-                                       args = yaml_spec['spec']['containers'][0]['args'],
-                                       volume_mounts = [k8s_client.V1VolumeMount(
-                                           name=yaml_spec['spec']['containers'][0]['volumeMounts'][0]['name'],
-                                           mount_path=yaml_spec['spec']['containers'][0]['volumeMounts'][0]['mountPath'],
-                                       )],
-                                       env = [k8s_client.V1EnvVar(
-                                           name=yaml_spec['spec']['containers'][0]['env'][0]['name'],
-                                           value=yaml_spec['spec']['containers'][0]['env'][0]['value'],
-                                       )])
+                                       args = yaml_spec['spec']['containers'][0]['args'])
```

Comment:
I'm a bit concerned about this change to the container-building functions for non-GKE deployments.
For container-building functions there might be no way for the user to supply secrets.

Reply:
Yes, I also think it's important to fix. However, I don't currently have bandwidth. Let's get this scheduled soon then.